pkg/client/apiutil/apimachinery.go

```diff
@@ -36,6 +36,9 @@ import (
 func NewDiscoveryRESTMapper(c *rest.Config) (meta.RESTMapper, error) {
 	// Get a mapper
 	dc := discovery.NewDiscoveryClientForConfigOrDie(c)
+
+	// TODO(mszostok): here we have useful info about all supported groups and resource
+	// in api-server
 	gr, err := restmapper.GetAPIGroupResources(dc)
```

Comment:
here we have info about all supported groups that we could use for checking if Lease is available, but it's not exposed, so I decided to use  the `RESTMapping` method from mapper and assert error via `IsNoMatchError`